### PR TITLE
Adding rules documentation

### PR DESCRIPTION
**Pull request by eddynaka**

### CODE REVIEW

File: docs/ValidationRules.md [hunk 1]
```diff
@@ -0,0 +1,642 @@
+# Rules
```
Comment (michaelcfanning) [inline docs/ValidationRules.md:1]: >Rules [](start = 2, length = 5)

Call this file BinSkimRules.md or simply RuleMessages.md

Comment (michaelcfanning) [reply docs/ValidationRules.md:1]: we could consider injecting the tool name, so 'BinSkim Rules'. BinSkimRules.md (with the injected tool name) would be a good default file name.

---
In reply to: [524812423](https://github.com/microsoft/binskim/pull/308#discussion_r524812423) [](ancestors = 524812423)

File: docs/ValidationRules.md [hunk 2]
```diff
@@ -0,0 +1,642 @@
...
+
+## Rule `BA3001.EnablePositionIndependentExecutable`
+
+### Description
+
+A Position Independent Executable (PIE) relocates all of its sections at load time, including the code section, if ASLR is enabled in the Linux kernel (instead of just the stack/heap).  This makes ROP-style attacks more difficult. This can be enabled by passing '-f pie' to clang/gcc.
+
+### Messages
+
+#### `Pass_Executable`: Error
```
Comment (michaelcfanning) [inline docs/ValidationRules.md:11]: >Error [](start = 24, length = 5)

this is a 'pass' message, not an error. i notice all the messages are emitted as 'Error'.

File: docs/ValidationRules.md (hunk 1 above)
Comment (eddynaka) [reply docs/ValidationRules.md]: just updated to BinskimRules.md

File: docs/ValidationRules.md (hunk 2 above)
Comment (eddynaka) [reply docs/ValidationRules.md]: Fixed!

**Pull request merged**